### DISCUSSION TITLE
Confused about the general concept of PNPM

### DISCUSSION THREAD

**Discussion by MrAmericanMike** [pnpm/pnpm#8553] [Q&A] [answered]
So I though that the idea of PNPM is that it creates symlinks instead of installing packages for every project in the `node_modules` folders.

So today I was doing some cleanup on some of my projects. As a rough idea lets say my folder with different code bases in it was about 20Gb. After going directory by directory (Project by project) deleting all the `node_modules` folders when I check again the size of the main projects folder it was down to about 5Gb.

How is this possible? I though I was just deleting symlinks.

Am I missing something?

**Comment by xty** [comment, marked as answer]
> So I though that the idea of PNPM is that it creates symlinks instead of installing packages for every project in the node_modules folders.

While pnpm is known for creating [hard links](https://en.wikipedia.org/wiki/Hard_link) (not symlinks) from the global store, this might not always be the case. A number of factors could come into play:
- Settings in your per-project, per-user, and global `npmrc` files. In particular, pay special attention to the [`package-import-method`](https://pnpm.io/npmrc#package-import-method) parameter.
- Whether your project directories reside on the same drive/partition as the global store. See: [Does pnpm work across multiple drives or filesystems?](https://pnpm.io/faq#does-pnpm-work-across-multiple-drives-or-filesystems)

A further possibility is that the `node_modules` directories are **NOT** actually consuming the 15GB of disk space you think it is. See: [Why does my node_modules folder use disk space if packages are stored in a global store?](https://pnpm.io/faq#why-does-my-node_modules-folder-use-disk-space-if-packages-are-stored-in-a-global-store) In short, the space that appears to have been saved is likely not claimed exclusively by files in `node_modules` but rather shared with the global store.

> How is this possible? I though I was just deleting symlinks.

In any event, you're definitely not just deleting symlinks.
